File: sorting/countingsort/countingsort.py
def counting_sort(arr, k):
    n = len(arr)
    B = [0] * n
    C = [0] * (k + 1)

    for i in range(n):
        C[arr[i]] += 1

    for i in range(1, k + 1):
        C[i] += C[i - 1]

    for i in range(n - 1, -1, -1):
        B[C[arr[i]] - 1] = arr[i]
        C[arr[i]] -= 1

    return B

if __name__ == "__main__":
    arr = [4, 7, 8, 1, 25]
    k = max(arr)
    sorted_arr = counting_sort(arr, k)

    print(", ".join(map(str, sorted_arr)))

File: sorting/countingsort/countingsort.cpp
#include <iostream>

void CountingSort(int arr[], int B[], int n, int k) {
    int C[k + 1] = {0};

    for (int i = 0; i < n; i++) {
        C[arr[i]]++;
    }

    for (int i = 1; i <= k; i++) {
        C[i] += C[i - 1];
    }

    for (int i = n - 1; i >= 0; i--) {
        B[C[arr[i]] - 1] = arr[i];
        C[arr[i]]--;
    }
}

int main() {
    int arr[] = {4, 7, 8, 1, 25};
    int n = sizeof(arr) / sizeof(arr[0]);
    int k = 25;
    int B[n];

    CountingSort(arr, B, n, k);

    for (int i = 0; i < n; ++i) {
        std::cout << B[i];
        if (i < n - 1) {
            std::cout << ", ";
        }
    }

    std::cout << std::endl;

    return 0;
}